janusgraph-core/src/main/java/org/janusgraph/core/JanusGraphFactory.java

```diff
@@ -313,10 +314,10 @@ private static final String getAbsolutePath(final File configParent, String file
         File storedir = new File(file);
         if (!storedir.isAbsolute()) {
             String newFile = configParent.getAbsolutePath() + File.separator + file;
-            log.debug("Overwrote relative path: was {}, now {}", file, newFile);
+            log.debug("Overwrote relative path: was {}, now {}", sanitizeAndLaunder(file), sanitizeAndLaunder(newFile));
```

Comment:
Hi @sjudeng , 
I am trying to solve same problem in my project.
As fortify says "An attacker could take advantage of this behavior to forge log entries or inject malicious content into the log." and by looking at you code how com parsing each char of your string and build the string is going to help us prevent injecting? 
Or is this method is just to suppress fortify scan?

Reply:
Hi @vin0010. As you note there are two issues, one is mitigating the risk of log forging and the other is resolving the Fortify finding. Currently the sanitize implementation is minimal and just [replaces newline characters](https://github.com/JanusGraph/janusgraph/blob/d12adfbf083f575fa48860daa37bfbd0e6095369/janusgraph-core/src/main/java/org/janusgraph/util/system/LoggerUtil.java#L27), though this could definitely be extended to be more robust. The second part of the implementation [recreates the string](https://github.com/JanusGraph/janusgraph/blob/d12adfbf083f575fa48860daa37bfbd0e6095369/janusgraph-core/src/main/java/org/janusgraph/util/encoding/StringEncoding.java#L73-L78), which was still needed to clear the Fortify finding in this case even after sanitizing.